test/test_force_exit.c

```diff
@@ -5,36 +5,40 @@
  * found in the LICENSE file.
  */
 
+#include <stdbool.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <assert.h>
 #include <emscripten.h>
+#include <emscripten/eventloop.h>
 
-int result = 0;
+bool done = false;
 
-void EMSCRIPTEN_KEEPALIVE success() {
-  printf("success? %d\n", result);
-  assert(result == 10);
+void success() {
+  printf("atexit: done=%d\n", done);
+  assert(done);
+  emscripten_force_exit(0);
 }
 
 void EMSCRIPTEN_KEEPALIVE later() {
   printf("later, now force an exit\n");
-  result += 10;
-  emscripten_force_exit(result);
+  done = true;
+  emscripten_force_exit(0);
```

Comment:
Do we need force_exit on both lines 20 and 26?

Reply:
Ooop, no good catch.